### DISCUSSION TITLE
Adding children to ContentSwitcher

### DISCUSSION THREAD

**Discussion by manforowicz** [Textualize/textual#4682] [Q&A] [answered]
Hello!

## Question
How can I add a new child to a `ContentSwitcher`, without the content switcher immediately displaying it below its current child?

## What I've tried
I tried calling `content_switcher.mount(new_child)`. However, this caused the new child to immediately appear below the currently displayed child of the content switcher.

I did notice that the content switcher goes back to showing only one child after I've switched `content_switcher.current` to a different child ID.

## Why I'm trying to do this
I'm making a utility to track nodes on a CAN bus. On the left side of my UI, I have a live list of nodes detected. Selecting an item should cause the content switcher on the right show info about the node. Nodes may appear or disappear at any time.

## Relevant discussion

#2751 Allowed `TabbedContent` to add new children.

Thanks!

**Comment by manforowicz** [comment]
I found a solution!

I can properly add a new child to a content switcher by first setting its display property:
```py
new_child = Foo()
new_child.display = False
content_switcher.mount(new_child)
```

Maybe it would be good to mention this in the documentation?

**Comment by TomJGooding** [comment, marked as answer]
You can instead use the `ContentSwitcher.add_content` method added in Textual v0.68 - see #4642.

However I spotted this is missing from the docs as they haven't been deployed for a few weeks. I've opened issue #4685 to flag this!

**Reply by manforowicz** [reply]
Thanks you!
